lib/TaskProcessing/ContextChatSearchProvider.php

```diff
@@ -0,0 +1,204 @@
+		if ($userId === null) {
+			throw new \RuntimeException('User ID is required to process the prompt.');
+		}
+
+		if (!isset($input['prompt']) || !is_string($input['prompt'])) {
+			throw new \RuntimeException('Invalid input, expected "prompt" key with string value');
+		}
+
+		if (!isset($input['limit']) || !is_numeric($input['limit'])) {
+			throw new \RuntimeException('Invalid input, expected "limit" key with number value');
+		}
+
+		if (
+			!isset($input['scopeType']) || !is_string($input['scopeType'])
+			|| !isset($input['scopeList']) || !is_array($input['scopeList'])
+			|| !isset($input['scopeListMeta']) || !is_string($input['scopeListMeta'])
+		) {
+			throw new \RuntimeException('Invalid input, expected "scopeType" key with string value, "scopeList" key with array value and "scopeListMeta" key with string value');
+		}
+
+		try {
+			ScopeType::validate($input['scopeType']);
+		} catch (\InvalidArgumentException $e) {
+			throw new \RuntimeException($e->getMessage(), intval($e->getCode()), $e);
+		}
+		if ($input['scopeType'] === ScopeType::SOURCE) {
+			throw new \InvalidArgumentException('Invalid scope type, source cannot be used to search');
+		}
+
+		// unscoped query
+		if ($input['scopeType'] === ScopeType::NONE) {
+			$response = $this->langRopeService->docSearch(
+				$userId,
+				$input['prompt'],
+				null,
+				null,
+				$input['limit'],
+			);
+			if (isset($response['error'])) {
+				throw new \RuntimeException('No result in ContextChat response. ' . $response['error']);
+			}
+			return $this->processResponse($userId, $response);
+		}
+
+		// scoped query
+		$scopeList = array_unique($input['scopeList']);
+		if (count($scopeList) === 0) {
+			throw new \RuntimeException('Empty scope list provided, use unscoped query instead');
+		}
+
+		if ($input['scopeType'] === ScopeType::PROVIDER) {
+			/** @var array<string> $scopeList */
+			$processedScopes = $scopeList;
+			$this->logger->debug('No need to index sources, querying ContextChat', ['scopeType' => $input['scopeType'], 'scopeList' => $processedScopes]);
+		} else {
+			// this should never happen
+			throw new \InvalidArgumentException('Invalid scope type');
+		}
+
+		if (count($processedScopes) === 0) {
+			throw new \RuntimeException('No supported sources found in the scope list, extend the list or use unscoped query instead');
+		}
+
+		$response = $this->langRopeService->docSearch(
+			$userId,
+			$input['prompt'],
+			$input['scopeType'],
+			$processedScopes,
+		);
+
+		return $this->processResponse($userId, $response);
+	}
+
+	/**
+	 * Validate and enrich sources JSON strings of the response
+	 *
+	 * @param string $userId
+	 * @param array $response
+	 * @return array{sources: list<string>}
+	 * @throws \RuntimeException
+	 */
+	private function processResponse(string $userId, array $response): array {
+		if (isset($response['error'])) {
+			throw new \RuntimeException('No result in ContextChat response: ' . $response['error']);
```

Comment:
wdyt of rephrasing the error message to something along "Error received in ContextChat document search request:"